src/main/java/uk/gov/hmcts/ccd/domain/service/accessprofile/filter/matcher/CaseIdMatcher.java

```diff
@@ -0,0 +1,23 @@
+package uk.gov.hmcts.ccd.domain.service.accessprofile.filter.matcher;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignment;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignmentFilteringResult;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseDetails;
+
+@Slf4j
+@Component
+public class CaseIdMatcher implements AttributeMatcher {
+
+    @Override
+    public boolean matchAttribute(RoleAssignmentFilteringResult result, CaseDetails caseDetails) {
+        RoleAssignment roleAssignment = result.getRoleAssignment();
+        log.debug("Apply filter on case id {} for role assignment {}",
```

Comment:
it would be good some better logging, for example we could log the role case id and the case case id to see if they match.
also it would be good a log line that says whether they matched or not.
Furthermore, I wouldn't use the word "filter"